### DISCUSSION TITLE
Website bug when loading /link | 404 NOT_FOUND

### DISCUSSION THREAD

**Discussion by Lancito01** [vercel/community#6428] [Help] [answered]
### Summary

Hey there! So I'm hosting a [website](https://lancito01.dev) on Vercel, and I'm using the tags <Link> and <Route> to move between different sites. The issue is, when I go from say, "/" to "/about" directly, meaning I type this into the address bar without going to the "main" site first, it doesn't load and it throws a 404 error. Why is this, and how can I fix it if possible?

### Example

https://lancito01.dev/about

### Steps to Reproduce

Go to https://lancito01.dev/about

**Comment by versecafe** [comment]
@Lancito01 It's hard to know without being able to see your code but I've experienced a similar issue when hosting flutter web on Vercel where direct navigation to anything but the top level throws a 404 but in app nav works, it's likely due to the way your SPA is handling routing when deployed to Vercel, a temporary workaround is to make a `vercel.json` like this
```json
{
  "routes": [
    { "src": "/[^.]+", "dest": "/", "status": 200 }
  ]
}
```
which will bounce direct navigations up to the top level `/`

**Reply by Lancito01** [reply]
My code should be in a public repo named "lancito01.dev" if you wanna check it out 🙏

**Reply by versecafe** [reply]
@Lancito01 your issue comes because your direct link is wrong it's not https://lancito01.github.io/about the correct link is https://lancito01.github.io/about_me

**Reply by versecafe** [reply]
also your links don'e need to end in .html they only need a .file ending for other types html is the default.

**Reply by Lancito01** [reply]
nono, that's my old website. i bought a domain in vercel to stop using the github one

**Reply by versecafe** [reply]
@Lancito01 How is your output directory configured on Vercel?

**Reply by Lancito01** [reply]
I didn't change any of the default settings

**Reply by Lancito01** [reply]
Bumping this 🙏 

I made as little research as I could on the subject, and I realized that this doesn't happen on localhost when I'm running the website locally for development purposes, so it definitely has to be a Vercel issue. Like I mentioned, I didn't change any of the "default settings" I guess you could call it, and the repo is on my GitHub linked [here](https://github.com/lancito01/lancito01.dev). Any help is appreciated <3

**Reply by Lancito01** [reply]
![image](https://github.com/vercel/community/assets/68301335/d454acc7-12bc-4a53-8599-7a58c897c6d0)
here's the error that shows when you try to go to for example, https://lancito01.dev/andybot

**Reply by Lancito01** [reply]
I could really use some help with this 😭 @versecafe I couldn't find anything on the docs either :/

**Reply by amyegan** [reply, marked as answer]
Thanks for sharing the repo, @Lancito01. I was able to do a test deployment and see the problem you described. I solved it by adding a `vercel.json` file in the root directory with the following contents
```json
{
  "rewrites": [{ "source": "/(.*)", "destination": "/index.html" }]
}
```

That gets direct URL routing to work with react-router. If you're interested, you can learn more about rewrite configuration at [vercel.com/docs/projects/project-configuration#rewrites](https://vercel.com/docs/projects/project-configuration#rewrites)

**Reply by Lancito01** [reply]
that's all it needs? and vercel will automatically detect the JSON?

good to know, I'll give it a shot and let you know, thank you 🙌 

edit: it worked ❤️
